framework/src/org/checkerframework/common/basetype/BaseTypeVisitor.java

```diff
@@ -1359,13 +1359,25 @@ public Void visitUnary(UnaryTree node, Void p) {
                 || (node.getKind() == Tree.Kind.POSTFIX_DECREMENT)
                 || (node.getKind() == Tree.Kind.POSTFIX_INCREMENT)) {
             AnnotatedTypeMirror varType = atypeFactory.getAnnotatedTypeLhs(node.getExpression());
-            // For postfix increments/decrements, the value type is incorrect due to the workaround
-            // in GenericAnnotatedTypeFactory.addComputedTypeAnnotations(Tree, AnnotatedTypeMirror, boolean)
-            // for the following bug:
-            // See Issue 867: https://github.com/typetools/checker-framework/issues/867
-            // This means could result in a false warning (false positive) in some cases and a lack
-            // of a warning in other cases (false negative).
-            AnnotatedTypeMirror valueType = atypeFactory.getAnnotatedType(node);
+            Tree newValue = node;
+            AnnotatedTypeMirror valueType = atypeFactory.getAnnotatedType(newValue);
+
+            // When the operation is postfix, valueType is a type of a value before the
+            // assignment. We override valueType with a type from a store, if it's provided.
+            if (node.getKind() == Tree.Kind.POSTFIX_DECREMENT
+                    || node.getKind() == Tree.Kind.POSTFIX_INCREMENT) {
+                CFAbstractStore<?, ?> store = atypeFactory.getStoreAfter(newValue);
+                if (store != null) {
+                    CFAbstractValue<?> value =
```

Comment:
Why can't you use code more similar to how the `GenericATF` worked below, e.g. use `Value as = getInferredValueFor(node.getExpression());` ?